core-bundle/contao/library/Contao/InsertTags.php

```diff
@@ -704,6 +704,18 @@ static function ($allowedTag)
 					$arrCache[$strTag] = $attribute?->getValue()[$elements[1]] ?? null;
 					break;
 
+				// Form confirmation message
+				case 'form_confirmation_message':
+					if ($request?->getSession()->getFlashBag()->has(Form::SESSION_CONFIRMATION_KEY))
```

Comment:
We should add a `$flags[] = 'attr';` before this line to ensure that the message gets HTML encoded correctly.

Reply:
I did it before, but since the confirmation message is an HTML (tinyMCE), it would output as follows:

```<p>Thank you for the message</p>```

instead of just

```Thank you for the message```

Due to HTML tags being encoded.

Reply:
I thought it’s just text. In this case this should be fine as is, but we need to make sure that the HTML itself is safe.

Reply:
How can we do that? I think it's filtered in the backend on save, isn't it?

Reply:
The HTML from the backend should be save (or from a trusted user), I was more concerned about the input from the frontend, but this should be fine now with 185f28934d6e5f2bfa219b647a0f12631a57af0e I think.